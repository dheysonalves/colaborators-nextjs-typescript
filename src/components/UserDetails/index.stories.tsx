import { ComponentMeta, ComponentStory } from '@storybook/react'
import UserDetails from '.'

export default {
	title: 'Layout/NavBar/UserDetails',
	component: UserDetails,
	args: {
		name: 'Luiz Zlochevsky',
		status: 'meus dados'
	}
} as ComponentMeta<typeof UserDetails>

export const Default: ComponentStory<typeof UserDetails> = (args) => (
	<UserDetails {...args} />
)
