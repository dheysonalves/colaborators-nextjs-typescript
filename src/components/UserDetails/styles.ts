import styled from '@emotion/styled'
import { theme } from '../../styles/theme/lightTheme'
import DynamicTypography from '../Typography'

export const UserAvatar = styled.div`
	display: flex;
	justify-content: center;
	border-radius: 80px;
	width: 32px;
	height: 32px;
	margin-right: 10px;
	padding: 9px;
	background-color: ${theme.light.primary.green};
	align-self: center;

	@media (max-width: 800px) {
		margin-right: 0px;
	}
`
export const Name = styled(DynamicTypography)`
	line-height: 100%;
	font-size: 14px;
	color: ${theme.light.neutral.black};
	text-transform: uppercase;
	font-weight: 400;
`
