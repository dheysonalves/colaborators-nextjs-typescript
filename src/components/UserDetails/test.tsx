import { render, screen } from '@testing-library/react'

import UserDetails from '.'

global.window.matchMedia = jest.fn().mockReturnValue({
	matches: true,
	addListener: () => {
		return
	},
	removeListener: () => {
		return
	}
})

describe('<UserDetails />', () => {
	it('should render the heading name and status', () => {
		const { container } = render(
			<UserDetails name="Luiz Zlochevsky" status="meus dados" />
		)

		expect(
			screen.getByRole('heading', { name: /Luiz Zlochevsky/i })
		).toBeInTheDocument()

		expect(
			screen.getByRole('heading', { name: /meus dados/i })
		).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
})
