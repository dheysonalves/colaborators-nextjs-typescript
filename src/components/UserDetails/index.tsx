import React from 'react'
import { UserAvatar, Name } from './styles'
import { BaseTypography, Paragraph } from '../Typography/base.index'
import { Container } from '../utils/Flex/flex.styles'
import getInitialsName from '../../utils/getInitials'
import Media from 'react-media'

interface UserDetailsProps {
	name: string
	status: string
}

const UserDetails = ({ name, status }: UserDetailsProps) => {
	return (
		<Container>
			<UserAvatar>
				<Name tag="p">{getInitialsName(name)}</Name>
			</UserAvatar>
			<Media
				query="(min-width: 1200px)"
				render={() => (
					<Container column>
						<BaseTypography tag="h1">{name}</BaseTypography>
						<Paragraph tag="h2">{status}</Paragraph>
					</Container>
				)}
			/>
		</Container>
	)
}

export default UserDetails
