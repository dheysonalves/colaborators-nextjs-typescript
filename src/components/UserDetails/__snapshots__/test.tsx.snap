// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetails /> should render the heading name and status 1`] = `
<div
  class="css-89ps4g"
>
  <div
    class="css-t2lkvl"
  >
    <p
      class="css-32r4px"
    >
      LZ
    </p>
  </div>
  <div
    class="css-1s9w40j"
  >
    <h1
      class="css-a8r7il"
    >
      Luiz Zlochevsky
    </h1>
    <h2
      class="css-ht8tt0"
    >
      meus dados
    </h2>
  </div>
</div>
`
